dbdialog/src/main/java/org/pentaho/ui/database/event/DataHandler.java

```diff
@@ -1320,6 +1337,18 @@ private void setConnectionSpecificInfo( DatabaseMeta meta ) {
       indexTablespaceBox.setValue( meta.getIndexTablespace() );
     }
 
+    // Strict Number(38) interpretation
+    if ( strictBigNumberInterpretaion != null ) {
+      // Check if oracle
+      if ( meta.getDatabaseInterface() instanceof OracleDatabaseMeta ) {
+        strictBigNumberInterpretaion.setVisible( true );
+        strictBigNumberInterpretaion.setChecked( meta.strictBigNumberInterpretation() );
+      } else {
+        strictBigNumberInterpretaion.setVisible( false );
+        strictBigNumberInterpretaion.setChecked( false );
```

Comment:
We shouldn't remove these lines because we should always reset the state of the checkbox. For example we select oracle, check out checkbox and then switch to another type of connection. So, the state of the checkbox will stay True for the non-oracle connection